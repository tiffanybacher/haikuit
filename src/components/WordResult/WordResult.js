import React from 'react';

const WordResult = (props) => {
  let syllables;

  if (props.syllables) {
    syllables = props.syllables.count;
  } else {
    syllables = 1
  }

  return(
    <article className="WordResult">
      {!props.error && <h4>{props.word}:
         <span className="syllables"> {syllables} syllables</span>
      </h4>}
      {props.error && <p>{props.error}</p>}
    </article>
  );
}

export default WordResult;