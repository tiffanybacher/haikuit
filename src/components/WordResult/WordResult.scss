.WordResult {
  padding: 10px 0 0 0;
 
  .syllables {
    font-style: italic;
  }
}