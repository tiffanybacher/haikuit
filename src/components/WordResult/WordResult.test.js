import React from 'react';
import { shallow } from 'enzyme';
import WordResult from './WordResult';

describe('WordResult', () => {
  let wordResult;
  let wrapper;

  beforeEach(() => {
    wordResult = { word: 'tenacity', syllables: {count: 4} };
    wrapper = shallow(<WordResult {...wordResult} />);
  });

  it('should match snapshot if there is a syllables prop', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should match snapshot if there is no syllables prop', () => {
    wordResult = { word: 'cat' };
    wrapper = shallow(<WordResult {...wordResult} />);

    expect(wrapper).toMatchSnapshot();
  });
});