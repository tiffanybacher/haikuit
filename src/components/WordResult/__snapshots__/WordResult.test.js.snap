// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordResult should match snapshot if there is a syllables prop 1`] = `
<article
  className="WordResult"
>
  <h4>
    tenacity
    :
    <span
      className="syllables"
    >
       
      4
       syllables
    </span>
  </h4>
</article>
`;

exports[`WordResult should match snapshot if there is no syllables prop 1`] = `
<article
  className="WordResult"
>
  <h4>
    cat
    :
    <span
      className="syllables"
    >
       
      1
       syllables
    </span>
  </h4>
</article>
`;
